### ISSUE TITLE
(feature) Segment by providing a list of emails in a textarea

### ISSUE THREAD

**Issue by arius28** [bramley/phplist-plugin-segment#16]
Hi and thank you for this very useful plugin.

I find myself needing the following functionality and thought I could share it here since I think your segmentation plugin could possibly implement this as a new option/feature. I can't find any similar functionality existing in core PHPlist or as a plugin. 

I want to give it a shot and make a pull request (if I manage it) but I wanted to put it out here just to get a comment if possible as to whether this is feasible or not. Perhaps I am not considering some serious obstacle?

So, the use case for me is that I have my subscribers database but I have clients (already subscribed) who come to exhibitions/seminars we hold and I want to send a specific newsletter only to those who attended. We hold a couple of those events every month or 2 so going in the database and adding attributes just to target these people seems like a tedious task and I also can't sit and isolate them by their email address one by one using the segmentation plugin.

I think if there was an extra option to segment based on a textarea that you fill in with email addresses (one per line) and only send to them it would be a much easier and quicker way. Of course, the addresses you fill in will need to be subscribed already and included in the lists you select in the campaign.

Thank you for your time and contributions that make PHPlist so much more powerful!

**Comment by bramley** [comment]
The simplest way to handle this is to import those subscriber email addresses into a new list (such as Exhibition ABC), then send to only that list. It can be a private list so that it doesn't appear on the subscriber's preferences page.

**Comment by bramley** [comment]
After taking a look at the code, it doesn't appear to be too difficult to add an extra operator (the existing ONE operator) for an email address, and provide a text area field to enter target email addresses. The campaign  should be sent to only those subscribers whose email address is in the text area.

The existing operators use a text line field so the new operator would be a bit different when displaying and when constructing the sql.

**Comment by arius28** [comment]
Thanks for the response and all the possible solutions!

I would like to avoid creating extra lists for each exhibition etc and end up with 50 lists after a year and more after that. I 'd like it a bit more dynamic. 

I have to say I didn't think of trying the REGEXP since I am talking about usually a 150-200 email addresses so I didn't think I could do it like that. Maybe it's not the most elegant solution but I just tried it and since the input field doesn't have any applied size limit (and the limit imposed by the browser seems to be nowhere near as low as the input I 'm giving) it actually worked pretty well. I inserted a pipe delimited string of 200 fake email addresses with some real ones interspersed to see if the string is handled from start till end and hit the calculate button and it showed me only the true addresses it found in the subscribers.

I can definitely work with that but since there are some other people working the newsletter system who would probably stare at me a bit awkward if I told them to do a collation like that with a pipe as a delimiter in a text editor I 'll also give it a shot based on your guidelines and if I manage I will make a pull request since my knowledge of Git is pretty limited but I am willing to learn!

Thanks again!
